src/proxy_app/telemetry/logger.py

```
@@ -0,0 +1,267 @@
+            rid = kwargs.get("litellm_call_id") or str(id(kwargs))
+            if rid in self._first_token_times and self._first_token_times[rid] == 0.0:
+                self._first_token_times[rid] = _now_ms()
+        except Exception:
+            pass
+
+    # --- async success hook (non-blocking) ---
+    async def async_log_success_event(self, kwargs, response_obj, start_time, end_time):
+        try:
+            self._ensure_writer()
+            await self._enqueue(kwargs, response_obj, start_time, end_time, status="ok", error=None)
+        except Exception:
```

Ruff (0.15.17)

[warning] Do not catch blind exception: `Exception`

(BLE001)